src/v/arch/manifest.cc

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "arch/manifest.h"
+
+#include "arch/error.h"
+#include "arch/logger.h"
+#include "bytes/iobuf.h"
+#include "bytes/iobuf_istreambuf.h"
+#include "bytes/iobuf_ostreambuf.h"
+#include "hashing/murmur.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "storage/ntp_config.h"
+
+#include <seastar/core/coroutine.hh>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <algorithm>
+#include <array>
+
+namespace arch {
+
+manifest::manifest()
+  : _ntp()
+  , _rev() {}
+
+manifest::manifest(model::ntp ntp, model::revision_id rev)
+  : _ntp(std::move(ntp))
+  , _rev(rev) {}
+
+remote_manifest_path manifest::get_manifest_path() const {
+    constexpr uint32_t bitmask = 0xF0000000;
```

Comment:
add a comment on the intent of this mask (erase some bits?) if so shouldn't it be `(~bitmask) & hash()` ... maybe i don't follow the `and` logic.